src/preact/bento-ce.js

```diff
@@ -105,3 +71,49 @@ if (typeof AMP !== 'undefined' && AMP.BaseElement) {
 }
 
 export {BaseElement};
+
+let ExtendableHTMLElement;
+
+/**
+ * @param {typeof CeBaseElement} BaseElement
+ * @return {typeof HTMLElement}
+ */
+function createBentoCustomElement(BaseElement) {
+  if (!ExtendableHTMLElement) {
+    ExtendableHTMLElement = maybeWrapNativeSuper(HTMLElement);
+  }
+  return class CustomElement extends ExtendableHTMLElement {
+    /** */
+    constructor() {
+      super();
+
+      /** @const {!CeBaseElement} */
+      this.implementation = new BaseElement(this);
+    }
+
+    /** */
+    connectedCallback() {
+      this.classList.add('i-amphtml-built');
+      this.implementation.mountCallback();
+      this.implementation.buildCallback();
+    }
+
+    /** */
+    disconnectedCallback() {
+      this.implementation.unmountCallback();
+    }
+
+    /** @return {Promise<*>} */
+    getApi() {
+      return this.implementation.getApi();
+    }
+  };
+}
+
+/**
+ * @param {string} tag
+ * @param {typeof BaseElement} BaseElement
+ */
+export function defineBentoElement(tag, BaseElement) {
+  customElements.define(tag, createBentoCustomElement(BaseElement));
```

Comment:
When we eventually fold `CustomElement`'s methods into `PreactBaseElement`, the `createBentoCustomElement` shouldn't even be necessary, right? we should be able to use `BaseElement` directly in `defineBentoElement`

Reply:
You might still something like:

```js
let ExtendableHTMLElement;
function createBentoCustomElement(BaseElement) {
  if (!ExtendableHTMLElement) {
    ExtendableHTMLElement = maybeWrapNativeSuper(HTMLElement);
  }
  return setSuperClass(BaseElement, ExtendableHTMLElement);
}
```

Reply:
to follow up on this, i don't think that will technically work because then the `BaseElement > PreactBaseElement` inheritance gets broken and transformed into `BaseElement > ExtendableHTMLElement`. instead i think this will work to ensure that `BaseElement > PreactBaseElement > ExtendableHTMLElement` :

You might still something like:

```javascript
let ExtendableHTMLElement;
function createBentoCustomElement(BaseElement) {
  if (!ExtendableHTMLElement) {
    ExtendableHTMLElement = maybeWrapNativeSuper(HTMLElement);
  }
  return setSuperClass(Object.getPrototypeOf(BaseElement), ExtendableHTMLElement);
}
```

Reply:
the drawback is that this will modify the PBE inheritance for everything that inherits off it, which would break in scenarios where a Bento component is on an AMP page..